utils/signing/signing.go

```diff
@@ -0,0 +1,154 @@
+package signing
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+type Ed25519Signer struct {
+	Public  ed25519.PublicKey  `json:"public"`
+	private ed25519.PrivateKey `json:"-"`
+	seed    []byte             `json:"-"`
+}
+
+func (k *Ed25519Signer) String() string {
+	return fmt.Sprintf("{Public: %v}", k.GetPublicKey())
+}
+
+func (k *Ed25519Signer) GoString() string {
+	return fmt.Sprintf("KeyPair(%q)", k.String())
+}
+
+func (k *Ed25519Signer) MarshalJSON() (jsonBytes []byte, err error) {
+	return []byte(fmt.Sprintf("{\"public\":%q}", k.GetPublicKey())), nil
+}
+
+func (k *Ed25519Signer) GetPublicKey() string {
+	return base64.StdEncoding.EncodeToString(k.Public)
+}
+
+func (k *Ed25519Signer) GetPrivateKey() string {
+	return base64.StdEncoding.EncodeToString(k.private)
+}
+
+func (k *Ed25519Signer) Sign(message []byte) (signature []byte, err error) {
```

Comment:
could you add a `GenerateSignature(message []byte)(base64EncodedSignature string, err error)` just to make it even slimpler to users